core/src/components/datetime/datetime.tsx

```diff
@@ -325,6 +325,7 @@ export class Datetime implements ComponentInterface {
           text: this.cancelText,
           role: 'cancel',
           handler: () => {
+            this.updateDatetimeValue(this.value);
```

Comment:
@liamdebeasi thats why i stored the changed value on an internal State in my PR (only as an info) :)

Reply:
Makes sense! I'd just rather reset it than store two copies of data that could potentially differ